### PR TITLE
zendesk helpcenter and yt playlist can be configured in each guild

### PR DESCRIPTION
**Pull request by JulianVennen**

### CODE REVIEW

File: commands/playlist.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      let service = google.youtube('v3');
+      let response = await service.playlists.list({
+        auth: config.googleapikey,
+        part: 'id',
+        id: playlist
+      });
+
+      if (response.data.items.length === 0) {
+        await message.react(util.icons.error);
+        await message.channel.send("Invlaid playlist!");
```
Comment (matthi4s) [inline commands/playlist.js:35]:
```suggestion
        await message.channel.send("Invalid playlist!");
```

File: commands/article.js [hunk 2]
```diff
@@ -1,16 +1,23 @@
 const config = require('../config.json');
+const util = require('../lib/util.js');
 const axios = require('axios');
 
 exports.command = async (message, args, database, bot) => {
 
+  let guildConfig = await util.getGuildConfig(message);
+  if (!guildConfig.helpcenter) {
+    await message.channel.send('No helpcenter configured!');
```
Comment (KurtThiemann) [inline commands/article.js:9]:
```suggestion
    await message.channel.send('No help center configured!');
```

File: commands/helpcenter.js [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+const axios = require('axios');
+
+exports.command = async (message, args, database, bot) => {
+    //Permission check
+    if (!message.member.hasPermission('MANAGE_GUILD')) {
+        await message.channel.send('You need the "Manage Server" permission to use this command.');
+        return;
+    }
+
+    let subdomain = /zendesk.com/.test(args[0]) ? args[0].split(/.zendesk.com/)[0] : args[0];
```
Comment (KurtThiemann) [inline commands/helpcenter.js:12]: Check if the string actually **ends** with **.**zendesk**.**com. Remove invalid characters from the subdomain (Zendesk subdomains can only include alphanumeric characters).

```suggestion
    let subdomain = args[0].replace(/\.zendesk\.com$/i, '').replace(/[^a-zA-Z\d]/g, '');
```

File: commands/helpcenter.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+    if (!subdomain) {
```
Comment (KurtThiemann) [inline commands/helpcenter.js:14]: Don't allow empty strings

```suggestion
    if (!subdomain || !subdomain.length) {
```

File: commands/helpcenter.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+      await message.channel.send("Please provide a zendesk.com adress or your zendesk subdomain!");
+      return;
+    }
+
+    if (["off","disabled","none"].includes(subdomain)) {
+      subdomain = null;
+    }
+    else {
+      try {
+        await axios.get(`https://${subdomain}.zendesk.com/api/v2/help_center/articles/search.json?query=a`);
```
Comment (KurtThiemann) [inline commands/helpcenter.js:25]: Don't use search endpoint

```suggestion
        await axios.get(`https://${subdomain}.zendesk.com/api/v2/help_center/articles.json`);
```

File: commands/playlist.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+const config = require('../config.json');
+
+exports.command = async (message, args, database, bot) => {
+    //Permission check
+    if (!message.member.hasPermission('MANAGE_GUILD')) {
+        message.channel.send('You need the "Manage Server" permission to use this command.');
+        return;
+    }
+
+    let playlist = /youtube.com/.test(args[0]) ? args[0].split(/list=/)[1].replace(/&[\w=]+/,'') : args[0];
```
Comment (KurtThiemann) [inline commands/playlist.js:14]:
```suggestion
    let match = String(args[0]).match(/youtube\.com\/.*[&?]list=(.+?)(?:&.*)?$/);
    let playlist = match ? match[1] : args[0];
```

File: commands/playlist.js [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+    if (!playlist) {
```
Comment (KurtThiemann) [inline commands/playlist.js:16]:
```suggestion
    if (!playlist || !playlist.length) {
```

File: commands/playlist.js [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+    }
+    else {
+      let service = google.youtube('v3');
+      let response = await service.playlists.list({
+        auth: config.googleapikey,
+        part: 'id',
+        id: playlist
+      });
+
+      if (response.data.items.length === 0) {
```
Comment (KurtThiemann) [inline commands/playlist.js:33]: Wouldn't that just mean that the playlist is empty and not necessarily that it is invalid?

Comment (JulianVennen) [reply commands/playlist.js:33]: No. I'm querying playlists not playlist items so it means that there is no playlist that matches the filters

File: commands/helpcenter.js [hunk 9]
```diff
@@ -0,0 +1,50 @@
...
+        return;
+    }
+
+    if (!args[0]) {
+      await message.react(util.icons.error);
+      await message.channel.send("Please provide a Zendesk subdomain e.g. https://example.zendesk.com");
+      return;
+    }
+
+    let subdomain = args.shift().replace(/\.zendesk\.com$/i, '').replace(/[^a-zA-Z\d]/g, '');;
```
Comment (KurtThiemann) [inline commands/helpcenter.js:18]: Also replace http(s)://

```suggestion
    let subdomain = args.shift().replace(/^https?:\/\/|\.zendesk\.com(\/.*)?$/ig, '').replace(/[^a-zA-Z\d]/g, '');;
```

**Pull request merged**